design/Events_design.md

```diff
@@ -0,0 +1,300 @@
+    /// @param amount0 The amount of token1 protocol fees that is withdrawn
+    event CollectProtocol(address indexed sender, address indexed recipient, uint128 amount0, uint128 amount1);
+```
+
+What all these events have in common is that the address topics like: `sender`, `recipient`, `owner`, `to`, etc, represent the 
+accounts which are affected by this transaction, and which are thus directly interested in it.
+
+
+##### Without end-user address fields
+
+```
+    /// @notice Emitted when a pool is created
+    /// @param token0 The first token of the pool by address sort order
+    /// @param token1 The second token of the pool by address sort order
+    /// @param fee The fee collected upon every swap in the pool, denominated in hundredths of a bip
+    /// @param tickSpacing The minimum number of ticks between initialized ticks
+    /// @param pool The address of the created pool
+    event PoolCreated(
+        address indexed token0,
+        address indexed token1,
+        uint24 indexed fee,
+        int24 tickSpacing,
+        address pool
+    );
+```
+
+```
+    /// @notice Emitted when a new fee amount is enabled for pool creation via the factory
+    /// @param fee The enabled fee, denominated in hundredths of a bip
+    /// @param tickSpacing The minimum number of ticks between initialized ticks for pools created with the given fee
+    event FeeAmountEnabled(uint24 indexed fee, int24 indexed tickSpacing);
+```
+
+```
+    event Sync(uint112 reserve0, uint112 reserve1);
+```
+
+```
+    /// @notice Emitted when the protocol fee is changed by the pool
+    /// @param feeProtocol0Old The previous value of the token0 protocol fee
+    /// @param feeProtocol1Old The previous value of the token1 protocol fee
+    /// @param feeProtocol0New The updated value of the token0 protocol fee
+    /// @param feeProtocol1New The updated value of the token1 protocol fee
+    event SetFeeProtocol(uint8 feeProtocol0Old, uint8 feeProtocol1Old, uint8 feeProtocol0New, uint8 feeProtocol1New);
+```
+
+What these events have in common is that they are not user-specific. They represent a general update of the smart contract.
+
+##### Visibility Rules
+
+Users should be able to request and read all events that are relevant to them.
+
+The rules we propose are:
+
+1. An event is considered relevant to all account owners whose addresses are used as topics in the event.
+2. In case there is no address in a topic, then the event is considered a lifecycle event, and thus relevant to everyone.
+3. An event is considered relevant to the signer of the transaction that created it. 
+
+The purpose for these rules is to be simple, clear, intuitive, and to work as good as possible with the existing contracts.  
+
+In case the rules above are not providing the desired functionality, the developer can adjust the topics.
+
+For example, if one of the lifecycle events should only be visible to the administrators, the developer can add that address as a topic.
```

Comment:
Let's clarify here that you cannot create a lifecycle event strictly visible to the administrators, since the tx submitter will always be able to see it.

Reply:
yes! very good point